File: frontend/src/api/recommendations.ts
import { apiRequest } from "./client";

export type TodayRecommendationItem = {
  title: string;
  description: string;
  reason: string;
  action_type: string;
  related_disease: string;
  priority: number;
};

export type TodayRecommendations = {
  date: string;
  items: TodayRecommendationItem[];
};

export async function getTodayRecommendations(): Promise<TodayRecommendations> {
  return apiRequest<TodayRecommendations>("/recommendations/today");
}

File: app/dtos/recommendations.py
from datetime import date

from pydantic import BaseModel


class TodayRecommendationItemResponse(BaseModel):
    title: str
    description: str
    reason: str
    action_type: str
    related_disease: str
    priority: int


class TodayRecommendationsResponse(BaseModel):
    date: date
    items: list[TodayRecommendationItemResponse]
